src/models/whatsnewmodel.cpp

```diff
@@ -0,0 +1,131 @@
+#include "featurelist.h"
+#include "feature.h"
+
+#include <QList>
+
+namespace {
+Logger logger(LOG_MODEL, "WhatsNewModel");
+}
+
+WhatsNewModel::WhatsNewModel() {
+  MVPN_COUNT_CTOR(WhatsNewModel);
+
+  WhatsNewModel::initialize();
+}
+
+WhatsNewModel::~WhatsNewModel() { MVPN_COUNT_DTOR(WhatsNewModel); }
+
+int WhatsNewModel::featureCount() const { return m_featurelist.count(); }
+
+QHash<int, QByteArray> WhatsNewModel::roleNames() const {
+  QHash<int, QByteArray> roles;
+  roles[RoleFeature] = "feature";
+
+  return roles;
+}
+
+int WhatsNewModel::rowCount(const QModelIndex&) const { return featureCount(); }
+
+QVariant WhatsNewModel::data(const QModelIndex& index, int role) const {
+  if (!index.isValid()) {
+    return QVariant();
+  }
+
+  switch (role) {
+    case RoleFeature:
+      return QVariant::fromValue(m_featurelist.at(index.row()));
+    default:
+      return QVariant();
+  }
+}
+
+void WhatsNewModel::initialize() {
+  QList<Feature*> allFeatures = Feature::getAll();
+  QList<Feature*> newFeatures;
+
+  for (int i = 0; i < allFeatures.count(); ++i) {
+    bool shouldBeInWhatsNew = allFeatures[i]->isNew() &&
+                              allFeatures[i]->isMajor() &&
+                              allFeatures[i]->isSupported();
+
+    if (shouldBeInWhatsNew) {
+      newFeatures.append(allFeatures[i]);
+    }
+  }
+
+  m_featurelist = newFeatures;
+}
+
+void WhatsNewModel::addSeenFeature(const QString& featureID) {
+  SettingsHolder* settingsHolder = SettingsHolder::instance();
+  Q_ASSERT(settingsHolder);
+
+  logger.debug() << "Add seen feature to settings";
+  if (!settingsHolder->hasSeenFeatures()) {
+    return;
+  }
+
+  QStringList seenfeatureslist;
+  seenfeatureslist = settingsHolder->seenFeatures();
+
+  if (seenfeatureslist.contains(featureID)) {
+    return;
+  }
+
+  seenfeatureslist.append(featureID);
+  settingsHolder->setSeenFeatures(seenfeatureslist);
+}
+
+void WhatsNewModel::removeSeenFeatures() {
+  SettingsHolder* settingsHolder = SettingsHolder::instance();
+  Q_ASSERT(settingsHolder);
+
+  logger.debug() << "Remove seen features from settings";
+  if (!settingsHolder->hasSeenFeatures()) {
+    return;
+  }
+
+  settingsHolder->setSeenFeatures(QStringList());
+}
+
+bool WhatsNewModel::hasUnseenFeature() {
+  SettingsHolder* settingsHolder = SettingsHolder::instance();
+  Q_ASSERT(settingsHolder);
+
+  logger.debug() << "Reading seen features from settings";
+  if (!settingsHolder->hasSeenFeatures()) {
+    return false;
+  }
+
+  const QStringList& seenFeatureList = settingsHolder->seenFeatures();
+  for (Feature* feature : m_featurelist) {
+    if (!seenFeatureList.contains(feature->id())) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+void WhatsNewModel::markFeaturesAsSeen() {
+  for (Feature* feature : m_featurelist) {
+    addSeenFeature(feature->id());
+  }
+
+  emit hasUnseenFeatureChanged();
+}
+
+void WhatsNewModel::markFeaturesAsUnseen() {
+  removeSeenFeatures();
```

Comment:
removeSeenFeatures() is not used elsewhere. Maybe you can remove this method and move its code here. What do you think?

Reply:
This method actually is not used at all :) remove it!